docs/lang/articles/misc/ndarray_android.md

```diff
@@ -0,0 +1,449 @@
+---
+sidebar_position: 10
+---
+
+# Run a Taichi Program using Ndarray on Android
+
+Taichi's JIT (Just In Time) module compiles a Taichi kernel to the compute shaders according to the specified backend (`arch=`) and executes these shaders in Taichi's JIT runtime. Taichi's AOT (Ahead of Time) module, however, builds and saves the necessary compute shaders so that you can write your own runtime at a later point to load and execute these shaders on a platform without a Python environment.
+
+Taking a simulation of celestial bodies' orbits as an example, this tutorial walks you through the process of running a Taichi program using Ndarray on Android.
+
+> - Taichi kernels and compute shaders are *not* one-to-one mapping. Each Taichi kernel can generate multiple compute shaders, the number *usually* in line with the number of the for loops in the kernel.
+> - [Taichi's AOT (Ahead Of Time) module](https://github.com/taichi-dev/taichi/issues/3642) is currently a proof of concept under development and subject to change in the future.
+
+## A definition of Ndarray
+
+Taichi provides a data container called Ndarray.  An Ndarray is a multidimensional container of elements of the same type and size; an element in an Ndarray is virtually a scalar or a tensor.
+
+### Ndarray shape
+
+Ndarray shape defines the Ndarray's layout; element shape defines the element's layout. For example:
+
+- An Ndarray with an Ndarray shape of [2, 1024] and an element shape of [1] is an array of 2 x 1,024 = 2,048 scalars.
+- An Ndarray with an Ndarray shape of [128, 128] and an element shape of [2, 4] is an array of  128 x 128 = 16,384 2 by 4 matrices.
+
+### Ndarray dimension
+
+The dimension here refers to the number of dimensions of an Ndarray. For example:
+
+- The dimension of an Ndarray with an Ndarray shape of [1, 2, 3] is three.
+- The dimension of an Ndarray with an Ndarray shape of [500] is one.
+
+### Benefit of Ndarray
+
+Each Ndarray has a fixed dimension but gives you the flexibility of changing its shape in accordance with its dimension.
```

Comment:
more accurate: changing its field shape in accordance with its field dimension

Reply:
Is field shape the same thing as Ndarray shape?

Reply:
I would rather say Ndarray array, then left bracket `field_dim` right bracket. @ailzhang ?